src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -451,9 +451,48 @@ public static ISet<INamespaceSymbol> GetUsingNamespacesInScope(this SemanticMode
                 }
             }
 
+            // for type constraints use declared accessibility of its member declaration (class, interface, delegate, method)
+            if (type.IsParentKind(SyntaxKind.TypeConstraint) &&
+                type.Parent.IsParentKind(SyntaxKind.TypeParameterConstraintClause))
+            {
+                var memberDeclaration = type.GetAncestor<MemberDeclarationSyntax>();
+                var declaredAccessibility = semanticModel.GetDeclaredSymbol(memberDeclaration, cancellationToken).DeclaredAccessibility;
+
+                return declaredAccessibility;
```

Comment:
Related to @sharwell's comment - what is `DeclaredAccessibility` for interface members?